packages/components/src/toolbar-icon-button/index.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ToolbarItem from '../toolbar-item';
+import IconButton from '../icon-button';
+
+function ToolbarIconButton( props, ref ) {
```

Comment:
Do we really need to have this distinction between the button and the icon button? I'm aware that there are some legacy considerations to take into account but it probably would be easier to maintain only one component moving forward.

See the `Button` & `IconButton` audit https://github.com/WordPress/gutenberg/issues/16541 and the following part in particular:

> ### Two options for Button and IconButton:
>
> **1. Keep Button and IconButton separate components**
> 
> Now that it seems like both components will have text and icon options in the component, here would be the main differences:
> 
> They handle the text label differently. Button is primarily for text (with an option for a supplementary icon) and IconButton is primarily for an Icon (with an option for supplementary text).
> For IconButton, text is hidden by default. When text is shown, it’s centered underneath the icon. When text is not shown, it's visible as a hover tooltip. There's always an icon shown.
> For Button, text is shown by default. There's always text shown. When an icon is shown, it's aligned to the left. There's no need for a tooltip because the text is always shown.
> We simplify the components by automatically handling the text label. The label is always the same whether it's shown or not. The components automatically handles the tooltip.
> We can consider renaming Button to TextButton.
> 
> **2. Combine Button and IconButton**
> 
> It could be a simpler experience for designers/developers if we combined the two components. Here is an example of a [component library](https://mineral-ui.com/components/button) that does this.
> 
> The main advantage is that the user of the system won't have to figure out which component to use, since it's all-in-one. The disadvantage is that it might be complex figuring out the different props, and the combination of certain props (e.g. isPrimary + isIconOnly). We might have to split the two components for technical reasons.

 @mtias @davewhitley

Reply:
Ideally, if `Button` supports both text and icon button, `ToolbarButton` should inherit that and we can get rid of this `__experimentalToolbarIconButton`.

Reply:
Linking related PR: #19193

Reply:
Removed `ToolbarIconButton` thanks to #19299 and #19193. `ToolbarButton` should be used instead.